ozaria/site/components/interactive/PageInteractive/draggableOrdering/index.vue

```diff
@@ -15,122 +16,116 @@
         required: true
       },
 
+      localizedInteractiveConfig: {
+        type: Object,
+        required: true
+      },
+
       interactiveSession: {
         type: Object,
         default: undefined
       },
 
       codeLanguage: {
-        type: String
+        type: String,
+        required: true
       }
     },
 
     data () {
       return {
         draggableGroup: Math.random().toString(),
 
-        promptSlots: [
-          { id: 1, text: 'one option' },
-          { id: 2, text: 'two option' },
-          { id: 3, text: 'three option' },
-          { id: 4, text: 'four option' }
-        ],
-
-        answerSlots: (new Array(4)).fill(undefined)
+        promptSlots: (this.localizedInteractiveConfig.elements || [])
+          .map(({ elementId, ...rest }) => ({
+            ...rest,
+            id: elementId
+          }))
       }
     },
 
-    mounted () {
-      const draggableUl = this.$refs['draggable-col']
-      Sortable.create(draggableUl, {
-        swap: true,
-        onChange: e => {
-          const temp = this.promptSlots[e.oldIndex]
-          this.promptSlots[e.oldIndex] = this.promptSlots[e.newIndex]
-          this.promptSlots[e.newIndex] = temp
+    computed: {
+      labels () {
+        return (this.localizedInteractiveConfig.labels || []).map((label) => {
+          if (typeof label === 'string') {
+            return { text: label }
+          }
+
+          return label
+        })
+      },
+
+      artUrl () {
+        if (this.interactive.defaultArtAsset) {
+          return getOzariaAssetUrl(this.interactive.defaultArtAsset)
         }
-      })
+
+        return undefined
+      }
     }
   }
 </script>
 
 <template>
-  <div class="draggable-ordering-container">
-    <base-interactive-title
-      :interactive="interactive"
-    />
-
-    <div class="prompt-row">
-      <div id='draggable-col' class='slots-container' key="draggable-col" ref="draggable-col">
-        <ul v-for="prompt in promptSlots" class="draggable-slot" :key="prompt.id">
-          <li>{{ prompt.text }}</li>
-        </ul>
-      </div>
-
-      <ordering-slots
-        v-model="answerSlots"
-
-        :draggable-group="draggableGroup"
-
-        class="ordering-column"
-      />
-
-      <div class="art-container">
-        <img
-          src="https://codecombat.com/images/pages/home/built_for_teachers1.png"
-          alt="Art!"
+  <base-interactive-layout
+    :interactive="interactive"
+    :art-url="artUrl"
+  >
+    <div class="draggable-ordering-content">
+      <draggable
```

Comment:
I believed the spec was to swap the card with another card instead of shifting all the cards up or down.
This is why I used the base library Sortable, as VueDraggable doesn't yet support swapping.
Was shifting what was intended?

Reply:
Ah that provides some context for why Sortable was there - I couldn't figure out why.  Will pick this up in the final UX phase.